marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/function-metadata.xsl

```diff
@@ -0,0 +1,152 @@
+            <xsl:otherwise><xsl:value-of select="concat('xs:',$in)"/></xsl:otherwise>
+        </xsl:choose>
+    </xsl:function>
+
+
+    <xsl:template match="/">
+        <xsl:apply-templates/>
+    </xsl:template>
+
+    <xsl:template match="mapping:function-defs">
+        <axsl:stylesheet version="2.0"
+                         extension-element-prefixes="xdmp"
+        >
+            <xsl:choose>
+                <xsl:when test="@type='xquery'">
+                    <axdmp:import-module href="{@location}" namespace="{@namespace}"/>
+                    <axdmp:using namespace="{@namespace}"/>
+                </xsl:when>
+                <xsl:otherwise>
+                    <xsl:namespace name="ext">http://marklogic.com/entity-services/extra-functions</xsl:namespace>
+                    <xsl:namespace name="map">http://marklogic.com/xdmp/map</xsl:namespace>
+                    <xsl:copy-of select="namespace::*[string(.)!='http://marklogic.com/entity-services/mapping']"/>
+                    <xsl:apply-templates select="mapping:function-def"/>
+                </xsl:otherwise>
+            </xsl:choose>
+        </axsl:stylesheet>
+    </xsl:template>
+
+    <xsl:template match="mapping:function-def">
+        <axsl:function name="{concat('ext:',@name)}">
+            <xsl:if test="mapping:return/@type and mapping:return/@type!='' and not(matches(mapping:return/@type,' '))">
+                <xsl:attribute name="as">
+                    <xsl:value-of select="this:mapTypeName(mapping:return/@type)"/>
+                </xsl:attribute>
+            </xsl:if>
+            <xsl:for-each select="mapping:parameters/mapping:parameter">
+                <axsl:param>
+                    <xsl:attribute name="name">
+                        <xsl:value-of select="@name"/>
+                    </xsl:attribute>
+                    <xsl:if test="@type and @type!='' and not(matches(@type,' '))">
+                        <xsl:attribute name="as">
+                            <xsl:value-of select="this:mapTypeName(@type)"/>
+                        </xsl:attribute>
+                    </xsl:if>
+                </axsl:param>
+            </xsl:for-each>
+            <axdmp:javascript-call>
+                <xsl:attribute name="name">
+                    <xsl:value-of select="@name"/>
+                </xsl:attribute>
+                <xsl:attribute name="href">
+                    <xsl:value-of select="$location"/>
+                </xsl:attribute>
+                <xsl:for-each select="mapping:parameters/mapping:parameter">
+                    <axsl:with-param axdmp:dialect="1.0-ml">
+                        <xsl:attribute name="name">
+                            <xsl:value-of select="@name"/>
+                        </xsl:attribute>
+                        <xsl:attribute name="select">
+                            <xsl:value-of select="concat('if (empty($', @name ,')) then null-node{} else $', @name)"/>
```

Comment:
@ryanjdew 
This seems to work fine. I didn't have to add xsl:choose element. Let me know if you see any issues with this.

Reply:
@srinathgit  Can you submit a separate PR first where this file is included as-is, with a comment indicating the version of ML that it came from? Also include the functions in xquery-lib as they currently are in ML.

We'll get that quickly merged into develop (can do a force merge if needed), and then submit your PR again with the changes to these OOTB stylesheets and functions.

The benefit is that our version history will then make it very clear what changes we made to the OOTB stuff.